### PR TITLE
fix(amplify-console-hosting): add missing proxy support

### PR DESCRIPTION
**Pull request by jhockett**
#### Description of changes
Honor proxy settings for `amplify publish`

#### Issue #, if available
Fixes https://github.com/aws-amplify/amplify-cli/issues/12460

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [x] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by danielleadams — CHANGES_REQUESTED** [review]
@jhockett is there a way to test this?

**Comment by ykethan** [comment]
Test this by setting up a ubuntu EC2 instance and [tiny proxy](http://tinyproxy.github.io/).
with security groups and tiny config file setup to allow my ip address and ports.

setup env variables as provided here: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-proxy.html in the local terminal. do not forget the no proxy env 
created a new app with hosting and tested publish.

on sucessfull publish 
image

when we dont allow a user ip for example: 
image

also will obeserve
image

**Review by sobolk — APPROVED** [review]

File: packages/amplify-console-hosting/src/utils/amplify-console-utils.js [hunk 1]
```diff
@@ -84,9 +85,12 @@ function waitJobToSucceed(job, amplifyClient) {
 }
 
 async function httpPutFile(filePath, url) {
+  const proxy = process.env.HTTP_PROXY || process.env.HTTPS_PROXY;
+  const proxyOption = proxy ? { agent: new ProxyAgent(proxy) } : {};
```
[the remark this answers is not in the document]
Comment (awsluja) [reply packages/amplify-console-hosting/src/utils/amplify-console-utils.js:89]: see https://github.com/aws-amplify/amplify-cli/pull/13012

**Review by awsluja — CHANGES_REQUESTED** [review]
looks like you have pulled in ^6.3.0 version but need to update invocations of the constructor

File: packages/amplify-console-hosting/src/utils/amplify-console-utils.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhockett) [reply packages/amplify-console-hosting/src/utils/amplify-console-utils.js]: Updated.

**Review by awsluja — DISMISSED** [review]

**Review by sobolk — APPROVED** [review]

File: packages/amplify-console-hosting/src/utils/amplify-console-utils.js [hunk 2]
```diff
@@ -84,9 +85,12 @@ function waitJobToSucceed(job, amplifyClient) {
 }
 
 async function httpPutFile(filePath, url) {
+  const proxy = process.env.HTTP_PROXY || process.env.HTTPS_PROXY;
+  const proxyOption = proxy ? { agent: new ProxyAgent() } : {};
```
Comment (sobolk) [inline packages/amplify-console-hosting/src/utils/amplify-console-utils.js:89]: can you copy comment from other places like this that explains why we're not consuming `proxy` ?

**Review by sobolk — APPROVED** [review]

**Review by awsluja — APPROVED** [review]

**Pull request merged**